### PR TITLE
fix(property-filters): materialized property value search

### LINKED ISSUE #8664: Person property autocomplete is buggy
## Bug description

PR #8414 moved the person property autocomplete from postgres to clickhouse, but with some outstanding bugs.

Raised by @macobo [here](https://github.com/PostHog/posthog/pull/8414#pullrequestreview-884457605):

> This query is broken in a few ways:
> 
> 1. We should be using materialized properties where possible
> 2. It doesn't handle person updates - meaning that the top value might not be the top value, but instead just the user who has been updated the most.

## Expected behavior

- Deleted people shouldn't show up in the search
- Changed properties shouldn't show up in the search
- We should be using materialized columns to make things faster and reduce load

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Fixes https://github.com/PostHog/posthog/issues/8664
- Uses materialized columns for person property value autocomplete
- Upgrades the event property value autocomplete to also use materialized columns
- Discards deleted rows from the query

## How did you test this code?

- Manually by opening a property search filter and seeing the right data is there
- Making sure all tests still run (to be double confirmed with CI)

### CODE REVIEW

**Comment by macobo** [comment]
Q: Do you think it would be possible to add this to the benchmark suite? 2 tests - one for materialized, one for not materialized property.

File: ee/clickhouse/models/property.py [hunk 1]
```diff
@@ -534,26 +534,36 @@ def box_value(value: Any, remove_spaces=False) -> List[Any]:
 
 
 def get_property_values_for_key(key: str, team: Team, value: Optional[str] = None):
```
Comment (macobo) [inline ee/clickhouse/models/property.py:536]: Can we move this function somewhere less generic? e.g. `ee/clickhouse/queries`?

Reason: This file is probably the most important file for querying anything regarding clickhouse, the most generic and important functions live here. This function is used exactly once, making the code harder to follow.

File: ee/clickhouse/models/property.py [hunk 2]
```diff
@@ -534,26 +534,36 @@ def box_value(value: Any, remove_spaces=False) -> List[Any]:
...
+    property_field, _ = get_property_string_expr("event", key, "%(key)s", "properties")
```
Comment (macobo) [inline ee/clickhouse/models/property.py:537]: Type checker failing here.

Not sure why tests aren't catching this bug though - worth taking a look at coverage?

You might want to take a look at `@test_with_materialized_columns` decorator for testing this.

**Comment by mariusandra** [comment]
@macobo thanks for the review! Ready for another look.

Most changes have been done, and left some comments and questions inline. I added 4 benchmarks as well. They're still running, so results pending, but happy to receive comments on the implementation.

**Review by macobo — APPROVED** [review]
Great work! :ship:

**Review by mariusandra — COMMENTED** [review]
forgot to submit those inline comments... :/

File: ee/clickhouse/models/property.py (hunk 2 above)
Comment (macobo) [reply ee/clickhouse/models/property.py]: Good enough for now! :)

**Pull request merged**